evmbin/src/info.rs

```diff
@@ -168,14 +175,15 @@ fn dump_state(state: &state::State<state_db::StateDB>) -> Option<pod_state::PodS
 /// Execute EVM with given `ActionParams`.
 pub fn run<'a, F, X>(
 	spec: &'a spec::Spec,
-	trie_spec: TrieSpec,
+	trie_spec: &'a TrieSpec,
```

Comment:
Is this change strictly required for something? It seems you have to clone the spec anyway, so maybe better take it by value than by reference.

Reply:
I've tried what you recommended but I haven't been able to implement it successfully as it opened a can of worms. This is where I got to https://github.com/paritytech/parity-ethereum/commit/ed5e6674fb08a9b34d61fd4ac996ec2cc7b1c062

<details><p>
   Compiling evmbin v0.1.0 (/home/ltfschoen/code/github/paritytech/parity-ethereum-master/evmbin)
error[E0504]: cannot move `params` into closure because it is borrowed
  --> evmbin/./src/info.rs:95:34
   |
94 |     run(spec, trie_spec, &params.gas, spec.genesis_state(), |mut client| {
   |                           ---------- borrow of `params.gas` occurs here
95 |         let result = match client.call(params, &mut trace::NoopTracer, &mut informant) {
   |                                        ^^^^^^ move into closure occurs here

error[E0504]: cannot move `tx_input` into closure because it is borrowed
   --> evmbin/./src/info.rs:153:43
    |
152 |     let result = run(&spec_checked, trie_spec, &tx_input.transaction.gas, &pre_state, |mut client| {
    |                                                 -------------------- borrow of `tx_input.transaction` occurs here
153 |         let result = client.transact(&env_info, tx_input.transaction, trace::NoopTracer, tx_input.informant);
    |                                                 ^^^^^^^^ move into closure occurs here

error[E0382]: capture of partially moved value: `tx_input`
   --> evmbin/./src/info.rs:153:43
    |
136 |     let TxInput { name, idx, spec, pre_state, post_root, env_info, trie_spec, .. } = tx_input;
    |                                                                    --------- value moved here
...
153 |         let result = client.transact(&env_info, tx_input.transaction, trace::NoopTracer, tx_input.informant);
    |                                                 ^^^^^^^^ value captured here after move
    |
    = note: move occurs because `tx_input.trie_spec` has type `ethcore::TrieSpec`, which does not implement the `Copy` trait

error: aborting due to 3 previous errors
</p></details>

Reply:
yeah, if you destructure `tx_input` you can't really move it around. just destructure `transaction` as well and pass it?

Reply:
Thanks, that worked, very slick! I've pushed changes in this commit 4b984745c33cd1751e71f0445e8210b0ffa7dedb